### PR TITLE
ddl: migrate test-infra to testify for ddl/foreign_key_test.go

### LINKED ISSUE #29103: migrate test-infra to testify for `ddl/foreign_key_test.go`

### PR DESCRIPTION
**Pull request by Mini256**
### What problem does this PR solve?

Issue Number: close #29103

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by Mini256** [comment]
/cc tisonkun xhebox 

Could you mind taking a look at this PR?

File: ddl/foreign_key_test.go [hunk 1]
```diff
@@ -111,35 +94,47 @@ func getForeignKey(t table.Table, name string) *model.FKInfo {
 	return nil
 }
 
-func (s *testForeignKeySuite) TestForeignKey(c *C) {
+func TestForeignKey(t *testing.T) {
+	t.Parallel()
```
Comment (hawkingrei) [inline ddl/foreign_key_test.go:98]: remove ```t.Parallel()```. we have decided to remove all ```t.Parallel()``` to avoid data race in test.

**Review by hawkingrei — APPROVED** [review]
LGTM

**Comment by hawkingrei** [comment]
@tangenta @tisonkun @xhebox  PTAL

**Review by xhebox — APPROVED** [review]

**Pull request merged**